release/rancher/rancher.go

```diff
@@ -311,6 +322,158 @@ func formatContentLine(line string) string {
 	return strings.TrimSpace(line)
 }
 
+func GenerateMissingImagesList(version string, concurrencyLimit int) ([]string, error) {
+	if !semver.IsValid(version) {
+		return nil, errors.New("version is not a valid semver: " + version)
+	}
+	const rancherWindowsImagesFile = "rancher-windows-images.txt"
+	const rancherImagesFile = "rancher-images.txt"
+	rancherWindowsImages, err := getRancherPrimeArtifact(version, rancherWindowsImagesFile)
+	if err != nil {
+		return nil, errors.New("failed to get rancher windows images: " + err.Error())
+	}
+	rancherImages, err := getRancherPrimeArtifact(version, rancherImagesFile)
+	if err != nil {
+		return nil, errors.New("failed to get rancher images: " + err.Error())
+	}
+	images := append(rancherWindowsImages, rancherImages...)
+
+	// create an error group with a limit to prevent accidentaly doing a DOS attack against our registry
+	ctx, cancel := context.WithCancel(context.Background())
+	errGroup, _ := errgroup.WithContext(ctx)
+	errGroup.SetLimit(concurrencyLimit)
+	missingImagesChan := make(chan string, len(images))
+	// auth tokens can be reused, but maps need a lock for reading and writing in go routines
+	repositoryAuths := map[string]string{}
+	repositoryAuthsLock := &sync.RWMutex{}
+
+	for _, imageAndVersion := range images {
+		splitImage := strings.Split(imageAndVersion, ":")
```

Comment:
Should we be safe and do a strings.Contains(s, ":") first, then a length check before we do the assignments below? Not sure if we can ever get a bad string.